### ISSUE TITLE
[text-field] Missing API to set aria-label to labelless TextField

### ISSUE THREAD

**Issue by yuriy-fix** [vaadin/flow-components#2160]
@OlliTietavainenVaadin commented on [Tue Oct 22 2019](https://github.com/vaadin/web-components/issues/1229)

In the case of a TextField without a label, there should still be a way to set the `aria-label` attribute for accessibility reasons. The internal implementation `input` element is not available from the component so it's not possible to add the attribute manually.

---

@alvarezguille commented on [Thu Jul 09 2020](https://github.com/vaadin/web-components/issues/1229#issuecomment-655973097)

Adding enhancement label as this use case was not considered in initial design.

---

@rolfsmeds commented on [Thu Dec 17 2020](https://github.com/vaadin/web-components/issues/1229#issuecomment-747421884)

`aria-labelledby` should be removed when no label is set: https://github.com/vaadin/vaadin-text-field/issues/562

**Comment by web-padawan** [comment]
Closing in favor of #2946 which contains a list of components to be covered.
